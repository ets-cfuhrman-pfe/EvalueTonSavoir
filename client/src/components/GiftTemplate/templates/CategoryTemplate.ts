import { TemplateOptions } from './types';
import QuestionContainer from './QuestionContainerTemplate';
import Title from './TitleTemplate';
import { Category } from 'gift-pegjs';

type CategoryOptions = TemplateOptions & Category;

export default function CategoryTemplate( { title }: CategoryOptions): string {
    return `${QuestionContainer({
        children: Title({
            type: 'Cat√©gorie',
            title: title
        })
    })}`;
}
