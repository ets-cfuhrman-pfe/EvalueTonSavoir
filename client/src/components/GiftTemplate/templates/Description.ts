import { TemplateOptions, Description as DescriptionType } from './types';
import QuestionContainer from './QuestionContainer';
import Title from './Title';
import textType from './TextType';
import { ParagraphStyle } from '../constants';
import { state } from '.';

type DescriptionOptions = TemplateOptions & DescriptionType;

export default function Description({ title, stem }: DescriptionOptions): string {
    return `${QuestionContainer({
        children: [
            Title({
                type: 'Description',
                title: title
            }),
            `<p style="${ParagraphStyle(state.theme)}">${textType({
                text: stem
            })}</p>`
        ]
    })}`;
}
