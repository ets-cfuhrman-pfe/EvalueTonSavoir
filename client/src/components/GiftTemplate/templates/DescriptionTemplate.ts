import { TemplateOptions } from './types';
import QuestionContainer from './QuestionContainerTemplate';
import Title from './TitleTemplate';
import { Description } from 'gift-pegjs';
import StemTemplate from './StemTemplate';

type DescriptionOptions = TemplateOptions & Description;

export default function DescriptionTemplate({ title, formattedStem}: DescriptionOptions): string {
    return `${QuestionContainer({
        children: [
            Title({
                type: 'Description',
                title: title
            }),
            StemTemplate({formattedStem}),
        ]
    })}`;
}
