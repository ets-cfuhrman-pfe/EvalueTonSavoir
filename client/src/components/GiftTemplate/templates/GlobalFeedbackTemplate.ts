import { TemplateOptions } from './types';
import {FormattedTextTemplate} from './TextTypeTemplate';
import { state } from '.';
import { theme } from '../constants';
import { TextFormat } from 'gift-pegjs';

type GlobalFeedbackOptions = TemplateOptions & TextFormat;

export default function GlobalFeedbackTemplate({ format, text }: GlobalFeedbackOptions): string {
    const Container = `
  position: relative;
  margin-top: 1rem;
  padding: 0 1rem;
  background-color: ${theme(state.theme, 'beige100', 'black400')};
  color: ${theme(state.theme, 'beige900', 'gray200')};
  border: ${theme(state.theme, 'beige300', 'black300')} ${state.theme === 'light' ? 1 : 2}px solid;
  border-radius: 6px;
  box-shadow: 0px 2px 5px ${theme(state.theme, 'gray400', 'black800')};
`;

    return (format && text)
        ? `<div style="${Container}">
        <p>${FormattedTextTemplate({format: format, text: text})}</p>
      </div>`
        : ``;
}
