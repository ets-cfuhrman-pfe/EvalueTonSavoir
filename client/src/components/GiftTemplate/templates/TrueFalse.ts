import { TemplateOptions, TextChoice, TrueFalse as TrueFalseType } from './types';
import QuestionContainer from './QuestionContainer';
import textType from './TextType';
import GlobalFeedback from './GlobalFeedback';
import MultipleChoiceAnswers from './MultipleChoiceAnswers';
import Title from './Title';
import { ParagraphStyle } from '../constants';
import { state } from '.';

type TrueFalseOptions = TemplateOptions & TrueFalseType;

export default function TrueFalse({
    title,
    isTrue,
    stem,
    trueFeedback: trueFeedback,
    falseFeedback: falseFeedback,
    globalFeedback
}: TrueFalseOptions): string {
    const choices: TextChoice[] = [
        {
            text: {
                format: 'moodle',
                text: 'Vrai'
            },
            isCorrect: isTrue,
            weight: null,
            feedback: isTrue ? trueFeedback : falseFeedback
        },
        {
            text: {
                format: 'moodle',
                text: 'Faux'
            },
            isCorrect: !isTrue,
            weight: null,
            feedback: !isTrue ? trueFeedback : falseFeedback
        }
    ];

    return `${QuestionContainer({
        children: [
            Title({
                type: 'Vrai/Faux',
                title: title
            }),
            `<p style="${ParagraphStyle(state.theme)}">${textType({
                text: stem
            })}</p>`,
            MultipleChoiceAnswers({ choices: choices }),
            GlobalFeedback({ feedback: globalFeedback })
        ]
    })}`;
}
