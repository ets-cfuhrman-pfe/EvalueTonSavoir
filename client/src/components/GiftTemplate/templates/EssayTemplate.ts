import { TemplateOptions } from './types';
import QuestionContainer from './QuestionContainerTemplate';
import Title from './TitleTemplate';
import {textType} from './TextTypeTemplate';
import GlobalFeedbackTemplate from './GlobalFeedbackTemplate';
import { ParagraphStyle, TextAreaStyle } from '../constants';
import { state } from '.';
import { EssayQuestion } from 'gift-pegjs';

type EssayOptions = TemplateOptions & EssayQuestion;

export default function EssayTemplate({ title, formattedStem, formattedGlobalFeedback }: EssayOptions): string {
    return `${QuestionContainer({
        children: [
            Title({
                type: 'Développement',
                title: title
            }),
            `<p style="${ParagraphStyle(state.theme)}">${textType(formattedStem)}</p>`,
            `<textarea class="gift-textarea" style="${TextAreaStyle(
                state.theme
            )}" placeholder="Entrez votre réponse ici..."></textarea>`, 
            (formattedGlobalFeedback && formattedGlobalFeedback.text) ?
                GlobalFeedbackTemplate(formattedGlobalFeedback) : ``
        ]
    })}`;
}
