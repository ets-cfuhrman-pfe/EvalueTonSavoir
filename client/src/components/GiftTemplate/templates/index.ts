import Category from './Category';
import DescriptionTemplate from './Description';
import Essay from './Essay';
import Matching from './Matching';
import MultipleChoice from './MultipleChoice';
import Numerical from './Numerical';
import ShortAnswer from './ShortAnswer';
import TrueFalse from './TrueFalse';
import Error from './Error';
import {
    GIFTQuestion,
    // Category as CategoryType,
    // Description as DescriptionType,
    MultipleChoice as MultipleChoiceType,
    Numerical as NumericalType,
    ShortAnswer as ShortAnswerType,
    // Essay as EssayType,
    TrueFalse as TrueFalseType,
    Matching as MatchingType,
    DisplayOptions
} from './types';

export const state: DisplayOptions = { preview: true, theme: 'light' };

export default function Template(
    { type, ...keys }: GIFTQuestion,
    options?: Partial<DisplayOptions>
): string {
    Object.assign(state, options);

    switch (type) {
        // Category, Description, Essay are not supported?
        // case 'Category':
        //     return Category({ ...(keys as CategoryType) });
        // case 'Description':
        //     return Description({
        //         ...(keys as DescriptionType)
        //     });
        case 'MC':
            return MultipleChoice({
                ...(keys as MultipleChoiceType)
            });
        case 'Numerical':
            return Numerical({ ...(keys as NumericalType) });
        case 'Short':
            return ShortAnswer({
                ...(keys as ShortAnswerType)
            });
        // case 'Essay':
        //     return Essay({ ...(keys as EssayType) });
        case 'TF':
            return TrueFalse({ ...(keys as TrueFalseType) });
        case 'Matching':
            return Matching({ ...(keys as MatchingType) });
        default:
            // TODO: throw error for unsupported question types?
            // throw new Error(`Unsupported question type: ${type}`);
            return ``;
    }
}

export function ErrorTemplate(text: string, options?: Partial<DisplayOptions>): string {
    Object.assign(state, options);

    return Error(text);
}

export {
    Category,
    DescriptionTemplate as Description,
    Essay,
    Matching,
    MultipleChoice,
    Numerical,
    ShortAnswer,
    TrueFalse,
    Error
};
