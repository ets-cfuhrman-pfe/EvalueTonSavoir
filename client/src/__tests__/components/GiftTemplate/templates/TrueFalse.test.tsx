import { render } from '@testing-library/react';
import TrueFalse from '../../../../components/GiftTemplate/templates';
import { TemplateOptions, TrueFalse as TrueFalseType } from '../../../../components/GiftTemplate/templates/types';

// Mock the nanoid function
jest.mock('nanoid', () => ({
    nanoid: jest.fn(() => 'mocked-id')
  }));

const plainTextMock: TemplateOptions & TrueFalseType = {
    type: 'TF',
    hasEmbeddedAnswers: false,
    title: 'Sample True/False Title',
    stem: { format: 'plain', text: 'Sample Stem' },
    isTrue: true,
    trueFeedback: { format: 'plain', text: 'Correct!' },
    falseFeedback: { format: 'plain', text: 'Incorrect!' },
    globalFeedback: { format: 'plain', text: 'Sample Global Feedback' }
};

const katexMock: TemplateOptions & TrueFalseType = {
    type: 'TF',
    hasEmbeddedAnswers: false,
    title: 'Sample True/False Title',
    stem: { format: 'html', text: '$$\\frac{zzz}{yyy}$$' },
    isTrue: true,
    trueFeedback: { format: 'moodle', text: 'Correct!' },
    falseFeedback: { format: 'html', text: 'Incorrect!' },
    globalFeedback: { format: 'markdown', text: 'Sample Global Feedback' }
};

const moodleMock: TemplateOptions & TrueFalseType = {
    type: 'TF',
    hasEmbeddedAnswers: false,
    title: 'Sample True/False Title',
    stem: { format: 'moodle', text: 'Sample Stem' },
    isTrue: true,
    trueFeedback: { format: 'moodle', text: 'Correct!' },
    falseFeedback: { format: 'moodle', text: 'Incorrect!' },
    globalFeedback: { format: 'moodle', text: 'Sample Global Feedback' }
};

const imageMock: TemplateOptions & TrueFalseType = {
    type: 'TF',
    hasEmbeddedAnswers: false,
    title: 'Sample Short Answer Title with Image',
    stem: { format: 'plain', text: 'Sample Stem with Image' },
    trueFeedback: { format: 'moodle', text: 'Correct!' },
    isTrue: true,
    falseFeedback: { format: 'moodle', text: 'Incorrect!' },
    globalFeedback: { format: 'plain', text: '<img src="https://via.placeholder.com/150" alt="Sample Image" />'  }
};

test('TrueFalse snapshot test with plain text', () => {
    const { asFragment } = render(<TrueFalse {...plainTextMock} />);
    expect(asFragment()).toMatchSnapshot();
});

test('TrueFalse snapshot test with katex', () => {
    const { asFragment } = render(<TrueFalse {...katexMock} />);
    expect(asFragment()).toMatchSnapshot();
});

test('TrueFalse snapshot test with moodle', () => {
    const { asFragment } = render(<TrueFalse {...moodleMock} />);
    expect(asFragment()).toMatchSnapshot();
});

test('TrueFalse snapshot test with image', () => {
    const { asFragment } = render(<TrueFalse {...imageMock} />);
    expect(asFragment()).toMatchSnapshot();
});