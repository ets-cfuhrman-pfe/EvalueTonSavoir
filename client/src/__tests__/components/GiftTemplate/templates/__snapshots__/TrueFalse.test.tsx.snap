// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrueFalse snapshot test with image 1`] = `
<DocumentFragment>
  &lt;section style="
  flex-wrap: wrap;
  position: relative;
  padding: 1rem 1rem;
  margin-bottom: 0.5rem;
  background-color: hsl(0, 0%, 100%);
  border: solid hsl(0, 0%, 100%) 2px;
  border-radius: 6px;
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
"&gt;
  &lt;div style="
  display: flex;
  font-weight: bold;
"&gt;
    &lt;span&gt;
      &lt;span style="
  color: #5271FF;
  "&gt;Sample Short Answer Title with Image&lt;/span&gt;
    &lt;/span&gt;
    &lt;span style="
  margin-left: auto;
  padding-left: 0.75rem;
  flex: none;
 margin-bottom: 10px;"&gt;
      &lt;span style="
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
  padding-left: 0.7rem;
  padding-right: 0.7rem;
  padding-top: 0.4rem;
  padding-bottom: 0.4rem;
  border-radius: 4px;
  background-color: hsl(0, 0%, 100%);
  color: hsl(180, 15%, 41%);
"&gt;Réponse courte&lt;/span&gt;
    &lt;/span&gt;
  &lt;/div&gt;

  &lt;div style="
  display: flex;
"&gt;
    &lt;span&gt;
      &lt;p style="
  color: hsl(0, 0%, 0%);
" class="present-question-stem"&gt;
                      Sample Stem with Image &lt;img src="https://example.com/cat.gif" alt="" class="markdown-image" style="--image-max-width: 400px"&gt;

                  &lt;/p&gt;
    &lt;/span&gt;
  &lt;/div&gt;

    &lt;div&gt;
      &lt;span style="
  color: hsl(0, 0%, 0%);
"&gt;Réponse: &lt;/span&gt;&lt;input class="gift-input" type="text" style="
  display: inline-block;
  padding: 0.375rem 0.75rem;
  line-height: 1.5;
  color: hsl(0, 0%, 16%);
  background-color: hsl(0, 0%, 100%);
  border: 1px solid hsl(0, 0%, 81%);
  border-radius: 0.25rem;
  box-shadow: rgba(0, 0, 0, 0.05) 0px 1px 2px 0px;
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  width: 90%;
" placeholder="A, B, C"&gt;
        &lt;/div&gt;
      &lt;div style="
  position: relative;
  margin-top: 1rem;
  padding: 0 1rem;
  background-color: hsl(43, 100%, 94%);
  color: hsl(43, 95%, 9%);
  border: hsl(36, 84%, 93%) 1px solid;
  border-radius: 6px;
  box-shadow: 0px 2px 5px hsl(0, 0%, 74%);
"&gt;
        &lt;p&gt;&lt;img src="https://via.placeholder.com/150" alt="Sample Image"&gt;&lt;/p&gt;
      &lt;/div&gt;&lt;/section&gt;
</DocumentFragment>
`;

exports[`TrueFalse snapshot test with katex 1`] = `
<DocumentFragment>
  &lt;section style="
  flex-wrap: wrap;
  position: relative;
  padding: 1rem 1rem;
  margin-bottom: 0.5rem;
  background-color: hsl(0, 0%, 100%);
  border: solid hsl(0, 0%, 100%) 2px;
  border-radius: 6px;
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
"&gt;
  &lt;div style="
  display: flex;
  font-weight: bold;
"&gt;
    &lt;span&gt;
      &lt;span style="
  color: #5271FF;
  "&gt;Sample True/False Title&lt;/span&gt;
    &lt;/span&gt;
    &lt;span style="
  margin-left: auto;
  padding-left: 0.75rem;
  flex: none;
 margin-bottom: 10px;"&gt;
      &lt;span style="
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
  padding-left: 0.7rem;
  padding-right: 0.7rem;
  padding-top: 0.4rem;
  padding-bottom: 0.4rem;
  border-radius: 4px;
  background-color: hsl(0, 0%, 100%);
  color: hsl(180, 15%, 41%);
"&gt;Vrai/Faux&lt;/span&gt;
    &lt;/span&gt;
  &lt;/div&gt;

  &lt;div style="
  display: flex;
"&gt;
    &lt;span&gt;
      &lt;p style="
  color: hsl(0, 0%, 0%);
" class="present-question-stem"&gt;
                      &lt;span class="katex-display"&gt;&lt;span class="katex"&gt;&lt;span class="katex-mathml"&gt;&lt;math xmlns="http://www.w3.org/1998/Math/MathML" display="block"&gt;&lt;mrow&gt;&lt;mfrac&gt;&lt;mrow&gt;&lt;mi&gt;z&lt;/mi&gt;&lt;mi&gt;z&lt;/mi&gt;&lt;mi&gt;z&lt;/mi&gt;&lt;/mrow&gt;&lt;mrow&gt;&lt;mi&gt;y&lt;/mi&gt;&lt;mi&gt;y&lt;/mi&gt;&lt;mi&gt;y&lt;/mi&gt;&lt;/mrow&gt;&lt;/mfrac&gt;&lt;/mrow&gt;\\frac{zzz}{yyy}&lt;/math&gt;&lt;/span&gt;&lt;span class="katex-html" aria-hidden="true"&gt;&lt;span class="base"&gt;&lt;span class="strut" style="height:1.988em;vertical-align:-0.8804em;"&gt;&lt;/span&gt;&lt;span class="mord"&gt;&lt;span class="mopen nulldelimiter"&gt;&lt;/span&gt;&lt;span class="mfrac"&gt;&lt;span class="vlist-t vlist-t2"&gt;&lt;span class="vlist-r"&gt;&lt;span class="vlist" style="height:1.1076em;"&gt;&lt;span style="top:-2.314em;"&gt;&lt;span class="pstrut" style="height:3em;"&gt;&lt;/span&gt;&lt;span class="mord"&gt;&lt;span class="mord mathnormal" style="margin-right:0.03588em;"&gt;yyy&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;span style="top:-3.23em;"&gt;&lt;span class="pstrut" style="height:3em;"&gt;&lt;/span&gt;&lt;span class="frac-line" style="border-bottom-width:0.04em;"&gt;&lt;/span&gt;&lt;/span&gt;&lt;span style="top:-3.677em;"&gt;&lt;span class="pstrut" style="height:3em;"&gt;&lt;/span&gt;&lt;span class="mord"&gt;&lt;span class="mord mathnormal" style="margin-right:0.04398em;"&gt;zzz&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;span class="vlist-s"&gt;​&lt;/span&gt;&lt;/span&gt;&lt;span class="vlist-r"&gt;&lt;span class="vlist" style="height:0.8804em;"&gt;&lt;span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;span class="mclose nulldelimiter"&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;
                  &lt;/p&gt;
    &lt;/span&gt;
  &lt;/div&gt;
&lt;span style="
  color: hsl(0, 0%, 0%);
"&gt;Choisir une réponse:&lt;/span&gt;
        &lt;div class='multiple-choice-answers-container'&gt;
          &lt;input class="gift-input" type="radio" id="idmocked-id" name="idmocked-id"&gt;
          
            &lt;label style="
          display: inline-block;
          padding: 0.2em 0 0.2em 0;
         
  color: hsl(0, 0%, 0%);
" for="idmocked-id"&gt;
            Vrai
            &lt;/label&gt;
          &lt;svg data-testid="correct-icon" style="
  vertical-align: text-bottom;
  display: inline-block;
  margin-left: 0.1rem;
  margin-right: 0.2rem;
 
    width: 1em;
    color: hsl(120, 39%, 54%);
  " role="img" aria-hidden="true" focusable="false" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"&gt;&lt;path fill="currentColor" d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"&gt;&lt;/path&gt;&lt;/svg&gt;
          &lt;span class="feedback-container"&gt;Correct!&lt;/span&gt;
          &lt;/input&gt;
        &lt;/div&gt;
        
        &lt;div class='multiple-choice-answers-container'&gt;
          &lt;input class="gift-input" type="radio" id="idmocked-id" name="idmocked-id"&gt;
          
            &lt;label style="
          display: inline-block;
          padding: 0.2em 0 0.2em 0;
         
  color: hsl(0, 0%, 0%);
" for="idmocked-id"&gt;
            Faux
            &lt;/label&gt;
          &lt;svg data-testid="incorrect-icon" style="
  vertical-align: text-bottom;
  display: inline-block;
  margin-left: 0.1rem;
  margin-right: 0.2rem;
 
    width: 0.75em;
    color: hsl(2, 64%, 58%);
  " role="img" aria-hidden="true" focusable="false" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 352 512"&gt;&lt;path fill="currentColor" d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"&gt;&lt;/path&gt;&lt;/svg&gt;
          &lt;span class="feedback-container"&gt;Incorrect!&lt;/span&gt;
          &lt;/input&gt;
        &lt;/div&gt;
        &lt;div style="
  position: relative;
  margin-top: 1rem;
  padding: 0 1rem;
  background-color: hsl(43, 100%, 94%);
  color: hsl(43, 95%, 9%);
  border: hsl(36, 84%, 93%) 1px solid;
  border-radius: 6px;
  box-shadow: 0px 2px 5px hsl(0, 0%, 74%);
"&gt;
        &lt;p&gt;Sample Global Feedback&lt;/p&gt;
      &lt;/div&gt;&lt;/section&gt;
</DocumentFragment>
`;

exports[`TrueFalse snapshot test with moodle 1`] = `
<DocumentFragment>
  &lt;section style="
  flex-wrap: wrap;
  position: relative;
  padding: 1rem 1rem;
  margin-bottom: 0.5rem;
  background-color: hsl(0, 0%, 100%);
  border: solid hsl(0, 0%, 100%) 2px;
  border-radius: 6px;
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
"&gt;
  &lt;div style="
  display: flex;
  font-weight: bold;
"&gt;
    &lt;span&gt;
      &lt;span style="
  color: #5271FF;
  "&gt;Sample True/False Title&lt;/span&gt;
    &lt;/span&gt;
    &lt;span style="
  margin-left: auto;
  padding-left: 0.75rem;
  flex: none;
 margin-bottom: 10px;"&gt;
      &lt;span style="
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
  padding-left: 0.7rem;
  padding-right: 0.7rem;
  padding-top: 0.4rem;
  padding-bottom: 0.4rem;
  border-radius: 4px;
  background-color: hsl(0, 0%, 100%);
  color: hsl(180, 15%, 41%);
"&gt;Vrai/Faux&lt;/span&gt;
    &lt;/span&gt;
  &lt;/div&gt;

  &lt;div style="
  display: flex;
"&gt;
    &lt;span&gt;
      &lt;p style="
  color: hsl(0, 0%, 0%);
" class="present-question-stem"&gt;
                      Sample Stem
                  &lt;/p&gt;
    &lt;/span&gt;
  &lt;/div&gt;
&lt;span style="
  color: hsl(0, 0%, 0%);
"&gt;Choisir une réponse:&lt;/span&gt;
        &lt;div class='multiple-choice-answers-container'&gt;
          &lt;input class="gift-input" type="radio" id="idmocked-id" name="idmocked-id"&gt;
          
            &lt;label style="
          display: inline-block;
          padding: 0.2em 0 0.2em 0;
         
  color: hsl(0, 0%, 0%);
" for="idmocked-id"&gt;
            Vrai
            &lt;/label&gt;
          &lt;svg data-testid="correct-icon" style="
  vertical-align: text-bottom;
  display: inline-block;
  margin-left: 0.1rem;
  margin-right: 0.2rem;
 
    width: 1em;
    color: hsl(120, 39%, 54%);
  " role="img" aria-hidden="true" focusable="false" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"&gt;&lt;path fill="currentColor" d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"&gt;&lt;/path&gt;&lt;/svg&gt;
          &lt;span class="feedback-container"&gt;Correct!&lt;/span&gt;
          &lt;/input&gt;
        &lt;/div&gt;
        
        &lt;div class='multiple-choice-answers-container'&gt;
          &lt;input class="gift-input" type="radio" id="idmocked-id" name="idmocked-id"&gt;
          
            &lt;label style="
          display: inline-block;
          padding: 0.2em 0 0.2em 0;
         
  color: hsl(0, 0%, 0%);
" for="idmocked-id"&gt;
            Faux
            &lt;/label&gt;
          &lt;svg data-testid="incorrect-icon" style="
  vertical-align: text-bottom;
  display: inline-block;
  margin-left: 0.1rem;
  margin-right: 0.2rem;
 
    width: 0.75em;
    color: hsl(2, 64%, 58%);
  " role="img" aria-hidden="true" focusable="false" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 352 512"&gt;&lt;path fill="currentColor" d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"&gt;&lt;/path&gt;&lt;/svg&gt;
          &lt;span class="feedback-container"&gt;Incorrect!&lt;/span&gt;
          &lt;/input&gt;
        &lt;/div&gt;
        &lt;div style="
  position: relative;
  margin-top: 1rem;
  padding: 0 1rem;
  background-color: hsl(43, 100%, 94%);
  color: hsl(43, 95%, 9%);
  border: hsl(36, 84%, 93%) 1px solid;
  border-radius: 6px;
  box-shadow: 0px 2px 5px hsl(0, 0%, 74%);
"&gt;
        &lt;p&gt;Sample Global Feedback&lt;/p&gt;
      &lt;/div&gt;&lt;/section&gt;
</DocumentFragment>
`;

exports[`TrueFalse snapshot test with plain text 1`] = `
<DocumentFragment>
  &lt;section style="
  flex-wrap: wrap;
  position: relative;
  padding: 1rem 1rem;
  margin-bottom: 0.5rem;
  background-color: hsl(0, 0%, 100%);
  border: solid hsl(0, 0%, 100%) 2px;
  border-radius: 6px;
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
"&gt;
  &lt;div style="
  display: flex;
  font-weight: bold;
"&gt;
    &lt;span&gt;
      &lt;span style="
  color: #5271FF;
  "&gt;Sample True/False Title&lt;/span&gt;
    &lt;/span&gt;
    &lt;span style="
  margin-left: auto;
  padding-left: 0.75rem;
  flex: none;
 margin-bottom: 10px;"&gt;
      &lt;span style="
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
  padding-left: 0.7rem;
  padding-right: 0.7rem;
  padding-top: 0.4rem;
  padding-bottom: 0.4rem;
  border-radius: 4px;
  background-color: hsl(0, 0%, 100%);
  color: hsl(180, 15%, 41%);
"&gt;Vrai/Faux&lt;/span&gt;
    &lt;/span&gt;
  &lt;/div&gt;

  &lt;div style="
  display: flex;
"&gt;
    &lt;span&gt;
      &lt;p style="
  color: hsl(0, 0%, 0%);
" class="present-question-stem"&gt;
                      Sample Stem
                  &lt;/p&gt;
    &lt;/span&gt;
  &lt;/div&gt;
&lt;span style="
  color: hsl(0, 0%, 0%);
"&gt;Choisir une réponse:&lt;/span&gt;
        &lt;div class='multiple-choice-answers-container'&gt;
          &lt;input class="gift-input" type="radio" id="idmocked-id" name="idmocked-id"&gt;
          
            &lt;label style="
          display: inline-block;
          padding: 0.2em 0 0.2em 0;
         
  color: hsl(0, 0%, 0%);
" for="idmocked-id"&gt;
            Vrai
            &lt;/label&gt;
          &lt;svg data-testid="correct-icon" style="
  vertical-align: text-bottom;
  display: inline-block;
  margin-left: 0.1rem;
  margin-right: 0.2rem;
 
    width: 1em;
    color: hsl(120, 39%, 54%);
  " role="img" aria-hidden="true" focusable="false" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"&gt;&lt;path fill="currentColor" d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"&gt;&lt;/path&gt;&lt;/svg&gt;
          &lt;span class="feedback-container"&gt;Correct!&lt;/span&gt;
          &lt;/input&gt;
        &lt;/div&gt;
        
        &lt;div class='multiple-choice-answers-container'&gt;
          &lt;input class="gift-input" type="radio" id="idmocked-id" name="idmocked-id"&gt;
          
            &lt;label style="
          display: inline-block;
          padding: 0.2em 0 0.2em 0;
         
  color: hsl(0, 0%, 0%);
" for="idmocked-id"&gt;
            Faux
            &lt;/label&gt;
          &lt;svg data-testid="incorrect-icon" style="
  vertical-align: text-bottom;
  display: inline-block;
  margin-left: 0.1rem;
  margin-right: 0.2rem;
 
    width: 0.75em;
    color: hsl(2, 64%, 58%);
  " role="img" aria-hidden="true" focusable="false" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 352 512"&gt;&lt;path fill="currentColor" d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"&gt;&lt;/path&gt;&lt;/svg&gt;
          &lt;span class="feedback-container"&gt;Incorrect!&lt;/span&gt;
          &lt;/input&gt;
        &lt;/div&gt;
        &lt;div style="
  position: relative;
  margin-top: 1rem;
  padding: 0 1rem;
  background-color: hsl(43, 100%, 94%);
  color: hsl(43, 95%, 9%);
  border: hsl(36, 84%, 93%) 1px solid;
  border-radius: 6px;
  box-shadow: 0px 2px 5px hsl(0, 0%, 74%);
"&gt;
        &lt;p&gt;Sample Global Feedback&lt;/p&gt;
      &lt;/div&gt;&lt;/section&gt;
</DocumentFragment>
`;
